heat/core/types.py

```diff
@@ -854,6 +861,87 @@ def promote_types(
     return __type_promotions[typecode_type1][typecode_type2]
 
 
+def result_type(
+    *arrays_and_types: Tuple[Union[dndarray.DNDarray, Type[datatype], Any]]
+) -> Type[datatype]:
+    """
+    Returns the data type that results from type promotions rules performed in an arithmetic operation.
+
+    Parameters
+    ----------
+    arrays_and_types: List of arrays and types
+        Input arrays, types or numbers of the operation.
+
+    Examples
+    --------
+    >>> ht.result_type(ht.array([1], dtype=ht.int32), 1)
+    ht.int32
+    >>> ht.result_type(ht.float32, ht.array(1, dtype=ht.int8))
+    ht.float32
+    >>> ht.result_type("i8", "f4")
+    ht.float64
+    """
+
+    def result_type_rec(*arrays_and_types):
+        # derive type
+        arg = arrays_and_types[0]
+
+        try:
+            # array / tensor
+            if isinstance(arg, np.ndarray):
+                type1 = canonical_heat_type(arg.dtype.char)
+            else:
+                type1 = canonical_heat_type(arg.dtype)
+
+            if len(arg.shape) > 0:
+                prec1 = 0
+            else:
+                prec1 = 2
+        except Exception:
+            try:
+                # type
+                if isinstance(arg, np.dtype):
+                    arg = arg.char
+                type1 = canonical_heat_type(arg)
+                prec1 = 1
+            except Exception:
```

Comment:
why such a broad exception?

Reply:
what is raised when this happens? `TypeError`?

Reply:
Ah - I first thought you meant the except block and not the caught exception :facepalm: